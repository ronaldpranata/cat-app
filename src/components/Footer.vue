<template>
   <div id="footer">
       {{ t('copyright') }}
  </div>
</template>

<script>
import { defineComponent } from 'vue';
import { useI18n } from 'vue-i18n';

export default defineComponent({
  name: 'Footer',
  setup() {
    const { t } = useI18n();
    return { t };
  },
});
</script>
