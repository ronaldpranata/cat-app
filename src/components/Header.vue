<template>
    <div id="nav">
        <span v-for="(item,index) in links" :key="index">
            <router-link  :to="item.url">{{ t(item.name) }} </router-link> <span v-if="index!=(links.length-1)">|</span>
        </span>
    </div>
</template>

<script>
import { defineComponent } from 'vue';
import { useI18n } from 'vue-i18n';

export default defineComponent({
  name: 'Header',
  props: {
    links: Array,
  },
  setup() {
    const { t } = useI18n();
    return { t };
  },
});
</script>
<style scoped lang="scss">
#nav {
  padding: 30px;

  a {
    font-weight: bold;
    color: #2c3e50;

    &.router-link-exact-active {
      color: #42b983;
    }
  }
}
</style>
