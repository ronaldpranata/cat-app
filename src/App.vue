<template>
  <Header :links='links' />
  <router-view />
  <Footer />
</template>

<script>
// @ is an alias to /src
import Header from '@/components/Header.vue';
import Footer from '@/components/Footer.vue';

export default {
  name: 'App',
  components: {
    Header,
    Footer,
  },
  data() {
    return {
      links: [
        {
          url: '/',
          name: 'home',
        },
        {
          url: '/about',
          name: 'about',
        },
      ],
    };
  },
};
</script>

<style lang='scss'>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  .clickable{
    cursor: pointer;
  }
}
</style>
