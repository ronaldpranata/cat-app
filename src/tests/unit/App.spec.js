import { mount } from '@vue/test-utils';
import App from '@/App.vue';
import Header from '@/components/Header.vue';
import Footer from '@/components/Footer.vue';
import { createI18n } from 'vue-i18n';

const i18n = createI18n({
  // vue-i18n options here ...
});

const factory = () => mount(App, {
  global: {
    plugins: [i18n],
  },
});

describe('App', () => {
  test('renders a Header', () => {
    const wrapper = factory();
    const component = wrapper.findComponent(Header);
    expect(component.exists()).toBeTruthy();
  });

  test('renders a Footer', () => {
    const wrapper = factory();
    const component = wrapper.findComponent(Footer);
    expect(component.exists()).toBeTruthy();
  });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
