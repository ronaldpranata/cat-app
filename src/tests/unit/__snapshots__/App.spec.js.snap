// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div class="header">
  <div id="nav"><span><router-link to="/">home</router-link> <span>|</span></span><span><router-link to="/about">about</router-link> <!--v-if--></span></div><img alt="Vue logo" src="">
</div>
<router-view></router-view>
<div id="footer">copyright</div>
`;
